reconstruction/ecoli/dataclasses/process/external_pathway.py

```diff
@@ -0,0 +1,264 @@
+        Sum along the columns of the massBalance matrix to check for reaction mass balance
+        '''
+        return np.sum(self.balance_matrix, axis=0)
+
+    def _populate_derivative_and_jacobian(self):
+        '''Compile callable functions for computing the derivative.'''
+        self._make_derivative()
+
+        self._rates = build_ode.derivatives(self.symbolic_rates)
+        self._stoich_matrix = self.stoich_matrix()  # Matrix is small and can be cached for derivatives
+
+
+    def _make_y_dy(self):
+        '''This is building the right hand side of the ODE system using MM kinetics.'''
+
+        substrates = self.substrates
+        yStrings = ["y[%d]" % x for x in range(len(substrates))]
+        y = sp.symbols(yStrings)
+        rates = []
+        for colIdx in range(len(substrates)):
+            flux = self.kcats[colIdx] * y[colIdx]/ (self.kms[colIdx] + y[colIdx])
+            rates.append(flux)
+        return y, rates
+
+    def _make_derivative(self):
+        '''
+        Creates symbolic representation of the ordinary differential equations
+        and the Jacobian. Used during simulations.
+        '''
+        y, rates = self._make_y_dy()
+
+        rates = sp.Matrix(rates)
+        self.symbolic_rates = rates
+
+
+    def molecules_to_next_time_step(self, moleculeDict, enzymeDict, cellVolume,
+                                    nAvogadro, timeStepSec, method="LSODA", jit=True):
+
+        """
+        Calculates the changes in the counts of molecules in the next timestep
+        by solving an initial value ODE problem.
+
+        Args:
+            moleculeDict (dict): name and current counts of molecules
+                involved in the ODE
+            enzymeDict (dict): dictionary of enzyme counts
+            cellVolume (float): current volume of cell
+            nAvogadro (float): Avogadro's number
+            timeStepSec (float): current length of timestep in seconds
+            method (str): name of the ODE method to use
+            jit (bool): if True, use the jit compiled version of derivatives
+                functions
+
+
+        Returns:
+            moleculesNeeded (1d ndarray, ints): counts of molecules that need
+                to be consumed
+            allMoleculesChanges (1d ndarray, ints): expected changes in
+                molecule counts after timestep
+        """
+
+        subCounts, subNames, enzymeCounts, enzymeNames = [], [], [], []
+        for sub in self.substrates:
+            subCounts.append(moleculeDict[sub[0]])
+            subNames.append(sub[0])
+
+        for reaction in self.rxn_ids:
+            # if the reaction has an enzyme
+            if self.enzymes[reaction]:
+                enzymeCounts.append(enzymeDict[self.enzymes[reaction][0]])
+                enzymeNames.append(self.enzymes[reaction][0])
+            #this might need to be modelled differently
+            else:
+                enzymeCounts.append(1)
+                enzymeNames.append('None')
```

Comment:
It's unclear what's being assumed here. If a reaction does not have an enzyme, are we assuming that the reaction is spontaneous? Would setting the enzyme count to 1 ensure that this reaction does not bottleneck the pathway? I guess for vioAE we have enzymes for each reaction so this does not matter, but it would be great if the comment was a bit more specific on what will need to be done for this block if we have reactions without enzymes.

Reply:
Also, using the string `'None'` as a filler value for the list of enzyme names is a little bit risky. It looks like this list is not being used later in the function -- was this made just for debugging purposes?

Reply:
So actually the last reaction in the violacein pathway is spontaneous according to Metacyc https://biocyc.org/META/NEW-IMAGE?type=PATHWAY&object=PWY-7040&detail-level=2
 
This is something that I wanted to discuss with you actually because I am not sure how to model this situation. Maybe we can have a quick call or discuss via email.